http/src/main/java/io/micronaut/http/uri/UriMatchTemplate.java

```diff
@@ -37,7 +37,7 @@
  */
 public class UriMatchTemplate extends UriTemplate implements UriMatcher {
 
-    protected static final String VARIABLE_MATCH_PATTERN = "([^\\/\\?#&;\\+]";
+    protected static final String VARIABLE_MATCH_PATTERN = "([^\\/\\?#(?!\\{)&;\\+]";
```

Comment:
Values in URI can be prefixed by `#` sign and treated as variables which can be extracted from URI. Since expressions need to be prefixed by `#` as well, I've updated thes regex to treat only values prefixed by `#` and not immediately followed by `{` as variables